### PR TITLE
Brock Rohloff

### PR DESCRIPTION
**Pull request by TheBrockstar**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,5 +1,136 @@
...
+
+    // Unique Error Messages
+     const error500 = { errorMessage: "The post information could not be modified." };
+
+    // Error Handling Conditionals
+    if(!title && !contents) {
+        response.status(400).send(missingDataError);
+    }
+
+    if (!typeof title === typeof contents === 'string') {
```
[the remark this answers is not in the document]
Comment (TheBrockstar) [reply index.js]: It's not MVP. I wanted to practice using another conditional and thought that making sure that only a string is being passed to title and contents would be helpful. Type checking, essentially.

File: index.js [hunk 2]
```diff
@@ -1,5 +1,136 @@
...
+        response.status(400).send(invalidTypeError);
+    }
+
+    // Database Promise Method
+    db.insert({title, contents})
+    .then(postId => response.status(201).send(postId))
```
[the remark this answers is not in the document]
Comment (TheBrockstar) [reply index.js]: I misread that. That makes much more sense. You can see that I know how to do that though in the delete endpoint. If I have time after project today, I will go back and fix it!

File: index.js [hunk 3]
```diff
@@ -1,5 +1,136 @@
...
+server.put('/api/posts/:id', (request, response) => {
+
+    // Destructuring Request Body For Validation
+    const {title, contents} = request.body;
+
+    // Unique Error Messages
+     const error500 = { errorMessage: "The post information could not be modified." };
+
+    // Error Handling Conditionals
+    if(!title && !contents) {
```
[the remark this answers is not in the document]
Comment (TheBrockstar) [reply index.js]: Thank you! Before, I was doing the opposite (ie. (title && contents), and I must have forgotten to change the conditional.

File: index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (gooseandmegander) [reply index.js]: No worries! With `Brockstar`s like yourself, I have to nit-pick. 😛

File: index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (gooseandmegander) [reply index.js]: Gotcha, I thought so. I was confused/curious about the syntax. I haven't seen it like that before; I'd like to see it when you get it working! I passed a number through, and it didn't trigger.

**Pull request closed without merging**